ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/hooks/useWatchLastSnapshotForComplianceReports.ts

```diff
@@ -0,0 +1,90 @@
+import { useCallback } from 'react';
+
+import useInterval from 'hooks/useInterval';
+import { getRequestQueryStringForSearchFilter } from 'utils/searchUtils';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+import {
+    ComplianceReportSnapshot,
+    ComplianceScanConfigurationStatus,
+    fetchComplianceReportHistory,
+} from 'services/ComplianceScanConfigurationService';
+import useRestQuery from 'hooks/useRestQuery';
+
+async function fetchLastJobForConfiguration(
+    scanConfigId: string
+): Promise<ComplianceReportSnapshot | null> {
+    const query = getRequestQueryStringForSearchFilter({
+        'Report state': ['PREPARING', 'WAITING'],
+    });
+
+    const complianceReportSnapshots = await fetchComplianceReportHistory({
+        id: scanConfigId,
+        query,
+        page: 1,
+        perPage: 1,
+        showMyHistory: true,
+        sortOption: {
+            field: 'Report Completion Time',
+            reversed: true,
+        },
+    });
+
+    return complianceReportSnapshots[0] ?? null;
+}
+
+type ComplianceReportSnapshotLookup = Record<string, ComplianceReportSnapshot | null>;
```

Comment:
That's a good point. This would still work but if you've done something similar in other cases, I'd rather keep it consistent and do it the same way. The same reason we have multiple util functions dealing with string queries, I'd rather limit the variability of patterns so everything is more harmonious 👀

Reply:
There is a `tsconfig` setting that will make indexing into a `Record<string, T>` return a type of `T | undefined`, but when I check a few years ago there were a massive number of places we would have to change code so it wasn't something I pursued further.